### PR TITLE
Fix linting issues

### PR DESCRIPTION
**Pull request by allisonking**
# Purpose
https://github.com/ethyca/fidesops/pull/510 lints in CI, but we have many linting errors, so this PR addresses those

# Changes
Might be easiest to go commit by commit
- First run lint with the `--fix` flag for auto fixing
- Second tier of errors were unused variables/imports, so I manually went in and removed a bunch of things
- Third tier was miscellaneous, but mostly around destructuring 
- Also one more commit to fix the privacy center linting

Okay a few more which were only being found during an `npm run build`:
* Fixes error handling in the NewUserForm (before we were getting toasts that said 'undefined', but now the actual error message should show up. Incidentally, I had to log out the error to see the actual structure, since the payload in [the API docs](https://ethyca.github.io/fidesops/api/) does not appear to be accurate. I had to add some [type predicates](https://redux-toolkit.js.org/rtk-query/usage-with-typescript#inline-error-handling-example) to get this to work, which I'm not totally sure I did right.
* Fixes one other typing issue, which wasn't causing user noticeable errors, but was making typescript complain

And sorry in advance, my prettier automatically ran on some of these files after saving 🙈 

# Checklist
- [ ] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file in appropriate sections
- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

### CODE REVIEW

File: clients/admin-ui/src/features/user-management/DeleteUserModal.tsx [hunk 1]
```diff
@@ -32,19 +32,18 @@ function DeleteUserModal(user: User) {
       setConfirmValue(event.target.value);
     }
   };
+  const {id: userId, username} = user;
 
   const deletionValidation =
-    user.id &&
+    !!(userId &&
```
Comment (LKCSmith) [inline clients/admin-ui/src/features/user-management/DeleteUserModal.tsx:38]: Could this just be `userId`? Is there a need for the double !!

**Review by LKCSmith — APPROVED** [review]
Overall looks good - lots of files just changing single to double quotes 😆 Guess I should change that preference in my VSCode settings.

File: clients/admin-ui/src/features/user-management/DeleteUserModal.tsx (hunk 1 above)
Comment (allisonking) [reply clients/admin-ui/src/features/user-management/DeleteUserModal.tsx:38]: hm I honestly can't remember why I changed this 😅  the `!!` forces the type (which can be "" or undefined) to a boolean which can be useful, but doesn't look like it's required in this case

**Comment by allisonking** [comment]
> Overall looks good - lots of files just changing single to double quotes 😆 Guess I should change that preference in my VSCode settings.

Ah yeah, that's cause my editor was already using `prettier` so when I saved files it automatically ran 😭  but I think soon we'll standardize our prettier configs so hopefully you won't have to change much and it'll Just Work 😬

**Review by TheAndrewJackson — APPROVED** [review]
I think it looks good! We should add a `.prettierrc.json` config file soon like you mentioned so we all use the same formatting settings. In my branch I just switched everything to single quotes 😅  

That's an easy fix though! No big deal

**Comment by allisonking** [comment]
> I think it looks good! We should add a `.prettierrc.json` config file soon like you mentioned so we all use the same formatting settings. In my branch I just switched everything to single quotes 😅
> 
> That's an easy fix though! No big deal

Sorry for the trouble 😭 @elliotbonneville is gonna make a great [tooling library](https://github.com/ethyca/fidesui/issues/5) which will handle this! In the mean time we just have to be sad for a bit.

**Pull request merged**